source/server/admin/listeners_handler.cc

```diff
@@ -16,10 +16,24 @@ ListenersHandler::ListenersHandler(Server::Instance& server) : HandlerContextBas
 Http::Code ListenersHandler::handlerDrainListeners(absl::string_view url, Http::ResponseHeaderMap&,
                                                    Buffer::Instance& response, AdminStream&) {
   const Http::Utility::QueryParams params = Http::Utility::parseQueryString(url);
+
   ListenerManager::StopListenersType stop_listeners_type =
       params.find("inboundonly") != params.end() ? ListenerManager::StopListenersType::InboundOnly
                                                  : ListenerManager::StopListenersType::All;
-  server_.listenerManager().stopListeners(stop_listeners_type);
+
+  const bool graceful = params.find("graceful") != params.end();
+  if (graceful) {
+    if (server_.drainManager().drainClose()) {
```

Comment:
Is this always going to return true depending on the settings? I think you need an actual "drain sequence started" boolean?

In terms of what to do, I don't feel strongly about it as long as it's documented well. I think returning 200 OK is fine with me.

/wait

Reply:
Yeah actually this is going to flakily break on probabilistic drain. The underlying `draining_` variable already exists, do you think I should just add a getter to the drain manager interface? Hopefully won't need to change the parent DrainDecision class.

Reply:
Yeah I think a getter on the base interface is fine which would just say whether the drain sequence has started.

Reply:
Done.